av1an-core/src/vapoursynth.rs

```diff
@@ -210,59 +213,105 @@ pub fn create_vs_file(
       _ => return Err(anyhow!("invalid chunk method")),
     }
   )))?;
+  let chunk_method_lower = match chunk_method {
+    ChunkMethod::FFMS2 => "ffms2",
+    ChunkMethod::LSMASH => "lsmash",
+    ChunkMethod::DGDECNV => "dgdecnv",
+    ChunkMethod::BESTSOURCE => "bestsource",
+    _ => return Err(anyhow!("invalid chunk method")),
+  };
 
-  if chunk_method == ChunkMethod::DGDECNV {
-    // Run dgindexnv to generate the .dgi index file
-    let dgindexnv_output = temp.join("split").join("index.dgi");
-
-    Command::new("dgindexnv")
-      .arg("-h")
-      .arg("-i")
-      .arg(source)
-      .arg("-o")
-      .arg(&dgindexnv_output)
-      .output()?;
-
-    let dgindex_path = to_absolute_path(&dgindexnv_output)?;
-    load_script.write_all(
-      format!(
-        "from vapoursynth import core\n\
-              core.max_cache_size=1024\n\
-            core.dgdecodenv.DGSource(source={dgindex_path:?}).set_output()"
-      )
-      .as_bytes(),
-    )?;
-  } else if chunk_method == ChunkMethod::BESTSOURCE {
-    load_script.write_all(
-      format!(
-        "from vapoursynth import core\n\
-          core.max_cache_size=1024\n\
-        core.bs.VideoSource({source:?}, cachepath={cache_file:?}).set_output()"
-      )
-      .as_bytes(),
-    )?;
-  } else {
-    load_script.write_all(
-      // TODO should probably check if the syntax for rust strings and escaping utf and stuff like that is the same as in python
-      format!(
-        "from vapoursynth import core\n\
-            core.max_cache_size=1024\n\
-      core.{}(r\"{}\", cachefile={:?}).set_output()",
+  // Only used for DGDECNV
+  let dgindex_path = match chunk_method {
+    ChunkMethod::DGDECNV => {
+      let dgindexnv_output = temp.join("split").join("index.dgi");
+
+      // Run dgindexnv to generate the .dgi index file
+      Command::new("dgindexnv")
+        .arg("-h")
+        .arg("-i")
+        .arg(&source)
+        .arg("-o")
+        .arg(&dgindexnv_output)
+        .output()?;
+
+      &to_absolute_path(&dgindexnv_output)?
+    }
+    _ => &source,
+  };
+
+  // Include rich loadscript.vpy and specify source, chunk_method, and cache_file
+  // Also specify downscale_height, pixel_format, and scaler for Scene Detection
+  // TODO should probably check if the syntax for rust strings and escaping utf and stuff like that is the same as in python
+  let mut load_script_text = include_str!("loadscript.vpy")
+    .replace(
+      "source = os.environ.get('AV1AN_SOURCE', None)",
+      &format!(
+        "source = r\"{}\"",
         match chunk_method {
-          ChunkMethod::FFMS2 => "ffms2.Source",
-          ChunkMethod::LSMASH => "lsmas.LWLibavSource",
-          _ => unreachable!(),
-        },
-        source.display(),
-        cache_file
-      )
-      .as_bytes(),
-    )?;
+          ChunkMethod::DGDECNV => dgindex_path.display(),
+          _ => source.display(),
+        }
+      ),
+    )
+    .replace(
```

Comment:
I'm curious, can we do this by setting the respective environment variables instead of by replacing each of these in the text?